### ISSUE TITLE
[Python] Define meter APIs and implement in Python agent

### ISSUE THREAD

**Issue by kezhenxu94** [apache/skywalking#7084]
We focused on tracing in Python agent since it's created, now that the tracing plugins are steadily grow, it's good time to hand on metrics stuff, this is a fundamental feature in Python agent that many other metrics-related features may be built on top of, such as PVM metrics collection #5944 .

I know @gloria-veg will be working on this, let's track the task in this issue.

**Comment by gloria-veg** [comment]
Thanks, @kezhenxu94! I will be dedicated to this in the following months. (#^.^#)

**Comment by Superskyyy** [comment]
Hello, @gloria-veg is this task being actively worked on? I see there's another issue waiting on this one(#5944). It will be fabulous if we can make it happen and release it in 1.0.0.

I see there's concrete progress on your branch, just not sure about the current status.

**Comment by gloria-veg** [comment]
Hi superSky,

Thanks for your email! Yes, I am still working on this. The frame is ready but there are still some details that need to be adjusted and tested. I may not have enough time in the next two months to work on this because I am struggling to survive in the heavy workload of school courses at Georgia Tech.

I am still interested in making it happen, and if you do not mind maybe we can finish this in winter vacation! Also, you can contact my mentor Zhenxu Ke to discuss this :)

Best,

Yuanyuan



At 2021-11-10 14:22:30, "Superskyyy" ***@***.***> wrote:

Hello, @gloria-veg is this task being actively worked on? I see there's another issue waiting on this one(#5944). It will be fabulous if we can make it happen and release it in 1.0.0.

I see there's concrete progress on your branch, just not sure about the current status.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Superskyyy** [comment]
> Hi superSky, Thanks for your email! Yes, I am still working on this. The frame is ready but there are still some details that need to be adjusted and tested. I may not have enough time in the next two months to work on this because I am struggling to survive in the heavy workload of school courses at Georgia Tech. I am still interested in making it happen, and if you do not mind maybe we can finish this in winter vacation! Also, you can contact my mentor Zhenxu Ke to discuss this :) Best, Yuanyuan At 2021-11-10 14:22:30, "Superskyyy" ***@***.***> wrote: Hello, @gloria-veg is this task being actively worked on? I see there's another issue waiting on this one(#5944). It will be fabulous if we can make it happen and release it in 1.0.0. I see there's concrete progress on your branch, just not sure about the current status. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

@kezhenxu94 FYI

@gloria-veg Nice to hear back that you are interested in making it happen! The main obstacle in your branch I see is being 36 commits behind our upstream master, where among the commits, our contributors have changed quite a lot of structures, especially coding guidelines. 

Please do let us know when you got time to resume the work :) so that we can work together to sort out the complexities in merging the master branch into your own. 

Btw it looks like we are in the same timezone, I certainly feel for you because I'm also under heavy course & lab work daily. For me it is quite relieving to switch around and do something fun and recognized in the open-source community when I get way too much stress in schoolwork, I guess that's an important reason why I keep coming back. Anyway, I look forward to your return!

Kind regards

**Comment by Superskyyy** [comment]
@eus-lwq has expressed interest in completing this feature, so I'm assigning the task to her.

**Comment by eus-lwq** [comment]
> @eus-lwq has expressed interest in completing this feature, so I'm assigning the task to her.

Yes please!

**Comment by ancin** [comment]
Could we got python application memory or CPU usage about this latest skywalking-python？@eus-lwq

**Comment by kezhenxu94** [comment]
> Could we got python application memory or CPU usage about this latest skywalking-python？@eus-lwq

Not yet

**Comment by wu-sheng** [comment]
> Could we got python application memory or CPU usage about this latest skywalking-python？@eus-lwq

I don't think we have an update about this issue in the upstream.

**Comment by Superskyyy** [comment]
> Could we got python application memory or CPU usage about this latest skywalking-python？@eus-lwq

The feature is implemented and pending merge, will be released in the next version.
